aws-rds-dbclusterendpoint/src/test/java/software/amazon/rds/dbclusterendpoint/UpdateHandlerTest.java

```diff
@@ -0,0 +1,294 @@
+        proxy = new AmazonWebServicesClientProxy(logger, MOCK_CREDENTIALS, () -> Duration.ofSeconds(600).toMillis());
+        rdsClient = mock(RdsClient.class);
+
+        rdsProxy = mockProxy(proxy, rdsClient);
+    }
+
+    @AfterEach
+    public void tear_down() {
+        verify(rdsProxy.client(), atLeastOnce()).serviceName();
+        verifyNoMoreInteractions(rdsClient);
+    }
+
+    @Test
+    public void handleRequest_ModifyDbClusterEndpoint_NotFound() {
+
+        when(rdsProxy.client().modifyDBClusterEndpoint(any(ModifyDbClusterEndpointRequest.class)))
+                .thenThrow(DbClusterEndpointNotFoundException.builder().message(MSG_NOT_FOUND).build());
+
+        final CallbackContext context = new CallbackContext();
+
+        test_handleRequest_base(
+                context,
+                null,
+                () -> RESOURCE_MODEL_BUILDER_WITH_TAGS().build(),
+                () -> RESOURCE_MODEL_BUILDER_WITH_TAGS().build(),
+                expectFailed(HandlerErrorCode.NotFound)
+        );
+
+        verify(rdsProxy.client(), times(1)).modifyDBClusterEndpoint(any(ModifyDbClusterEndpointRequest.class));
+
+    }
+
+    @Test
+    public void handleRequest_ModifyThrowsRuntimeError() {
+
+        when(rdsProxy.client().modifyDBClusterEndpoint(any(ModifyDbClusterEndpointRequest.class)))
+                .thenThrow(new RuntimeException("Internal Failure"));
+
+        final CallbackContext context = new CallbackContext();
+
+        test_handleRequest_base(
+                context,
+                null,
+                () -> RESOURCE_MODEL_BUILDER_WITH_TAGS().build(),
+                () -> RESOURCE_MODEL_BUILDER_WITH_TAGS().build(),
+                expectFailed(HandlerErrorCode.InternalFailure)
+        );
+
+        verify(rdsProxy.client(), times(1)).modifyDBClusterEndpoint(any(ModifyDbClusterEndpointRequest.class));
+
+    }
+
+    @Test
+    public void handleRequest_SimpleTagUpdate_Success() {
+        when(rdsProxy.client().addTagsToResource(any(AddTagsToResourceRequest.class)))
+                .thenReturn(AddTagsToResourceResponse.builder().build());
+        when(rdsProxy.client().removeTagsFromResource(any(RemoveTagsFromResourceRequest.class)))
+                .thenReturn(RemoveTagsFromResourceResponse.builder().build());
+        when(rdsProxy.client().listTagsForResource(any(ListTagsForResourceRequest.class)))
+                .thenReturn(ListTagsForResourceResponse.builder().build());
+        when(rdsProxy.client().modifyDBClusterEndpoint(any(ModifyDbClusterEndpointRequest.class)))
+                .thenReturn(ModifyDbClusterEndpointResponse.builder().build());
+
+        final CallbackContext context = new CallbackContext();
+
+        test_handleRequest_base(
+                context,
+                () -> DB_CLUSTER_ENDPOINT_AVAILABLE,
+                () -> RESOURCE_MODEL_BUILDER().tags(TAG_LIST_ALTER).build(),
+                () -> RESOURCE_MODEL_BUILDER_WITH_TAGS().build(),
```

Comment:
Nit: using the same base builder with alternative tag lists would make it much more expressive:
```java
RESOURCE_MODEL_BUILDER().tags(TAG_LIST).build()
```